### DISCUSSION TITLE
Error in GetBcyl when running FIELDLINES

### DISCUSSION THREAD

**Discussion by nicallen93** [PrincetonUniversity/STELLOPT#324] [Q&A]
Hey All,

I've been running FIELDLINES with various VMEC inputs and an associated mgrid. And for certain VMEC inputs, I get a floating point exception when it's running the virtual casing. I'm using mpi, so I'll sometimes get this error multiple times at different points.

" ERROR in GetBcyl Detected
 R,PHI,Z  0.63750000000000007       0.11780972778797150       0.24843750000000003
 br,bphi,bz,myworkid   6.2203533836300479E-002 -0.56349244183328773       0.44479155889157729               22
Note: The following floating-point exceptions are signalling: IEEE_INVALID_FLAG IEEE_OVERFLOW_FLAG IEEE_DENORMAL
STOP ERROR in GetBcyl"

I get both underflow and overflow errors that vary with each point it gets hung up at. Any help with understanding/resolving this issue and completing the field line following would be greatly appreciated. Let me know what additional information would be helpful.

Thanks,
Nic

**Comment by lazersos** [comment]
This error is issued from the `GetBcyl` routine which performs the inverse lookup in VMEC. This is usually the sign that the underlying equilibrium was highly shaped, and most likely not converged with respect to the spectral content.  Can you provide an input file so your VMEC case and FIELDLINES run can be reproduced? If not please let us know the max M and N values in your equilibrium boundary, your `MPOL` and `NTOR` values along with your `NS_ARRAY` and `FTOL_ARRAY`.

**Comment by nicallen93** [comment]
This would make sense as the surfaces are highly shaped when compared to other cases that I have run. 

I cannot upload the mgrid file I used in the equilibrium calculation as it is too large (~100 MB compared to GitHub's 25MG limit), but I've uploaded the input vmec file and the corresponding wout file. If there is another avenue for me to send you the mgrid file, let me know.

[test_files_no_mgrid.zip](https://github.com/user-attachments/files/17501866/test_files_no_mgrid.zip)

**Comment by lazersos** [comment]
@nicallen93 I tested your file by using the -plasma flag, so no fields outside the VMEC domain were calculated.  The code ran fine, no crash.  What version of the code are you using? For example:
```
-----  GIT Repository  -----
   Repository: git@github.com:PrincetonUniversity/STELLOPT.git
   Branch:     feature/THRIFT_DKES_PENTA
   Version:    v4.00-121-ge21c
   Built-on:   30.10.2024 13:39:01
   Hash:       e21c9e143d416d4ad4be9b82320ecb74a5017e07
```
